sql/004_search.sql

```diff
@@ -441,25 +472,166 @@ END;
 $$ LANGUAGE PLPGSQL;
 
 
+/* cql_query_op -- Parses a CQL query operation, recursing when necessary
+     IN jsonb -- a subelement from a valid stac query
+     IN text -- the operator being used on elements passed in
+     RETURNS a SQL fragment to be used in a WHERE clause
+*/
+DROP FUNCTION IF EXISTS cql2_query;
+CREATE OR REPLACE FUNCTION cql2_query(j jsonb, recursion int DEFAULT 0) RETURNS text AS $$
+DECLARE
+args jsonb := j->'args';
+jtype text := jsonb_typeof(j->'args');
+op text := lower(j->>'op');
+arg jsonb;
+argtext text;
+argstext text[] := '{}'::text[];
+inobj jsonb;
+_numeric text := '';
+ops jsonb :=
+    '{
+        "eq": "%s = %s",
+        "lt": "%s < %s",
+        "lte": "%s <= %s",
+        "gt": "%s > %s",
+        "gte": "%s >= %s",
+        "le": "%s <= %s",
+        "ge": "%s >= %s",
+        "=": "%s = %s",
+        "<": "%s < %s",
+        "<=": "%s <= %s",
+        ">": "%s > %s",
+        ">=": "%s >= %s",
+        "like": "%s LIKE %s",
+        "ilike": "%s ILIKE %s",
```

Comment:
Curious if this is CQL2 oriented, using the CQL1 spec, it looked like the [op `LIKE` was used with a `nocase` flag](https://portal.ogc.org/files/96288#:~:text=J-,If%20NOCASE%20is%20TRUE,-%2C%20the%20comparison%20SHALL). If that's not hte case anymore, could you add a test that indicates correct usage (as expected by pgStac)?

Reply:
cql2 uses upper and lower pseudo function on the property. Examples added in the tests.